elastica/experimental/connection_contact_joint/generic_system_type_connection.py

```diff
@@ -0,0 +1,414 @@
+        point_system_one : numpy.ndarray
+            Describes for system one in the local coordinate system the translation from the node `index_one` (for rods)
+            or the center of mass (for rigid bodies) to the joint.
+            (default = np.array([0.0, 0.0, 0.0]))
+        point_system_two : numpy.ndarray
+            Describes for system two in the local coordinate system the translation from the node `index_two` (for rods)
+            or the center of mass (for rigid bodies) to the joint.
+            (default = np.array([0.0, 0.0, 0.0]))
+        rest_rotation_matrix: np.array
+            2D (3,3) array containing data with 'float' type.
+            Rest 3x3 rotation matrix from system one to system two at the connected elements.
+            If provided, the rest rotation matrix is enforced between the two systems throughout the simulation.
+            If not provided, `rest_rotation_matrix` is initialized to the identity matrix,
+            which means that a restoring torque will be applied to align the directors of both systems directly.
+            (default=None)
+        """
+        GenericSystemTypeFreeJoint.__init__(
+            self,
+            k=k,
+            nu=nu,
+            point_system_one=point_system_one,
+            point_system_two=point_system_two,
+        )
+        FixedJoint.__init__(
+            self, k=k, nu=nu, kt=kt, nut=nut, rest_rotation_matrix=rest_rotation_matrix
+        )
+
+    # Apply force is same as free joint
+    def apply_forces(self, system_one, index_one, system_two, index_two):
+        return GenericSystemTypeFreeJoint.apply_forces(
+            self, system_one, index_one, system_two, index_two
+        )
+
+    def apply_torques(self, system_one, index_one, system_two, index_two):
+        return FixedJoint.apply_torques(
+            self, system_one, index_one, system_two, index_two
+        )
+
+
+def compute_position_of_point(system: SystemType, point: np.ndarray, index: int):
+    """
+    Computes the position in the inertial frame of a point specified in the local frame of
+    the specified node of the rod.
+
+    Parameters
+    ----------
+    system: SystemType
+        System to which the point belongs.
+    point : np.array
+        1D (3,) numpy array containing 'float' data.
+        The point describes a position in the local frame relative to the inertial position of node
+        with index `index` and orientation of element with `index`.
+    index: int
+        Index of the node / element in the system.
+
+    Returns
+    -------
+    position : np.array
+        1D (3,) numpy array containing 'float' data.
+        Position of the point in the inertial frame.
+
+    Examples
+    --------
+    Compute position in inertial frame for a point (0, 0, 1) relative to the last node of the rod.
+
+    >>> system.compute_position_of_point(np.array([0, 0, 1]), -1)
+    """
+    assert point.shape == (3,), "Point must be in the shape (3,)"
```

Comment:
Maybe not a good idea to have an `assert` in a function that will be called every time step?

Reply:
Removed in https://github.com/GazzolaLab/PyElastica/pull/149/commits/a01136f57363afdc4b835ad163283cbd8926b343